cudaproxy/src/cudaproxy.rs

```diff
@@ -51,6 +51,29 @@ pub extern "C" fn __cudaRegisterFatBinary(fatCubin: &FatHeader) -> *mut u64 {
     }
     return result as *mut u64;
 }
+#[no_mangle]
+pub extern "C" fn __cudaUnregisterFatBinary(fatCubinHandle:u64) {
+    // println!("Hijacked __cudaUnregisterFatBinary(fatCubinHandle = {:x})", fatCubinHandle);
+    let fatCubinPtr: *const u64 = fatCubinHandle as *const u64;
+    // unsafe{
+    // println!("Hijacked __cudaUnregisterFatBinaryEnd( the content of fatCubinHandle = {:x})", *fatCubinPtr);
+    // }
+
+    unsafe{
+        //  if *fatCubinPtr != 0 {
+        //  println!("the content of fatCubin poninter is not 0, need to unload the module");
+         syscall2(SYS_PROXY, ProxyCommand::CudaUnRegisterFatBinary as usize,fatCubinHandle as usize);
+        //  }
+    }
+}
+
+#[no_mangle]
+pub extern "C" fn __cudaRegisterFatBinaryEnd(fatCubinHandle:u64) {
```

Comment:
This one is not working? Then remove it.

Reply:
is working, it actually be called after register function API, otherwise segmentation fault